docs/development/machine-bootstrap.md

```diff
@@ -0,0 +1,19 @@
+# Machine-Controller integration into Gardener
+The Gardener is deploying the [machine-controller-manager](https://github.com/gardener/machine-controller-manager) as part of a Shoot controlplane onto a Seed to manage the nodes/machines of the Shoot cluster. The machine-controller-manager will create, bootstrap, update and delete the nodes of the Shoot cluster and the underlying virtual machines.
+
+## Flow of creating and bootstrapping Machines/Nodes
+The gardener-controller-manager will ensure during the creation/reconciliation flow
+- that the machine-controller-manager gets deployed into the Shoot namespace of the Seed.
+- that proper ``<cloudprovider>machineclass`` objects and its corresponding secrets exists in the Shoot namespace of the Seed. Those secrets contain information to bootstrap the node and will be copied to the machines of the related class.
+- that ``machine-deployments`` for the in the Shoot manifest specified worker groups exists. Those deployments will spawn the required ``machine`` objects in the Shoot namespace of the Seed.
+- that a valid bootstrap-token secret exists in the kube-system namespace of the Shoot.
+- that the cloud-config configuration for setting up nodes exists in the kube-system namespace of the Shoot.
+
+The machine-controller-manager will pick up the ``machine`` objects and create the vms on the respective infrastructure provider. The bootstrap of a vm, which should join the Shoot cluster as node, is done via CoreOS [cloud-config](https://coreos.com/os/docs/latest/cloud-config.html).
+
+The cloud-config configuration will be downloaded from the Shoot apiserver. This will be archieved by copying the contents of the machineclass secret onto the machine. This secret contains configuration to register a systemd service ``cloud-config-downloader`` and a kubeconfig to fetch only the cloud-config and the bootstrap-token from the Shoot apiserver. The downloaded cloud-config configuration will register additional systemd services for the kubelet and other required services on the node.
```

Comment:
s/cloud-config configuration/cloud-config

Reply:
s/apiserver/API server

Reply:
s/This will be archieved/This is achieved

Reply:
Explain that the machine class secret does not contain the "real" cloud-config but only the very small script/kubeconfig to download it.

--> s/will register.../does contain the actual cloud-config which contains all the services and configurations for the VM to register as Kubernetes node and to function properly in our scenario.

or sth similar